File: AmiClean-Backend/AmiClean.Domain/Entities/Notifikacija.cs
namespace AmiClean.Domain.Entities;

public class Notifikacija
{
    public int ID_Notifikacije { get; set; }
    public int FK_Korisnik { get; set; }
    public int? FK_Narudzba { get; set; }
    public string Kanal { get; set; } = null!;
    public string Naslov { get; set; } = null!;
    public string Poruka { get; set; } = null!;
    public DateTime Datum_Slanja { get; set; }
    public string Status_Slanja { get; set; } = "Na_cekanju";
    public bool Procitano { get; set; }

    public virtual Korisnik Korisnik { get; set; } = null!;
    public virtual Narudzba? Narudzba { get; set; }
}

File: AmiClean-App/lib/features/notifikacije/models/notifikacija.dart
class Notifikacija {
  const Notifikacija({
    required this.id,
    this.narudzbaId,
    required this.naslov,
    required this.poruka,
    required this.datumSlanja,
    required this.procitano,
  });

  final int id;
  final int? narudzbaId;
  final String naslov;
  final String poruka;
  final DateTime datumSlanja;
  final bool procitano;

  factory Notifikacija.fromJson(Map<String, dynamic> json) {
    return Notifikacija(
      id: json['id'] as int,
      narudzbaId: json['narudzbaId'] as int?,
      naslov: json['naslov'] as String,
      poruka: json['poruka'] as String,
      datumSlanja: DateTime.parse(json['datumSlanja'] as String),
      procitano: json['procitano'] as bool,
    );
  }
}
